tests/test_coadds.py

```diff
@@ -0,0 +1,335 @@
+                    patch_outer_bbox, psfSigma=cls.psf_sigmas[identifiers.cell]
+                )
+
+                for (inst_flux, position, shape) in sources:
+                    dataset.addSource(inst_flux, position, shape)
+
+                # Create a spatially varying variance plane.
+                variance = ImageD(
+                    np.random.uniform(
+                        0.8,
+                        1.2,
+                        (
+                            nx * cls.inner_size + 2 * cls.border_size,
+                            nx * cls.inner_size + 2 * cls.border_size,
+                        ),
+                    ),
+                    xy0=outer_bbox.getMin(),
+                )
+                exposure, _ = dataset.realize(variance.getArray() ** 0.5, schema, randomSeed=123456789)
+                cls.exposures[identifiers.cell] = exposure
+                exposure = exposure[outer_bbox]
+                image_plane = OwnedImagePlanes(
+                    image=exposure.image, variance=exposure.variance, mask=exposure.mask
+                )
+
+                single_cell_coadds.append(
+                    SingleCellCoadd(
+                        outer=image_plane,
+                        psf=SingleGaussianPsf(
+                            cls.psf_size, cls.psf_size, cls.psf_sigmas[Index2D(x=x, y=y)]
+                        ).computeKernelImage(outer_bbox.getCenter()),
+                        inner_bbox=geom.Box2I(
+                            geom.Point2I(x * cls.inner_size, y * cls.inner_size),
+                            geom.Extent2I(cls.inner_size, cls.inner_size),
+                        ),
+                        inputs={
+                            None,
+                        },
+                        common=common,
+                        identifiers=identifiers,
+                    )
+                )
+
+        grid_bbox = geom.Box2I(geom.Point2I(0, 0), geom.Extent2I(nx * cls.inner_size, ny * cls.inner_size))
+        grid = UniformGrid(grid_bbox, Index2D(x=nx, y=ny))
+
+        cls.multiple_cell_coadd = MultipleCellCoadd(
+            single_cell_coadds,
+            grid=grid,
+            outer_cell_size=geom.Extent2I(cls.outer_size, cls.outer_size),
+            inner_bbox=None,
+            common=common,
+            psf_image_size=geom.Extent2I(cls.psf_size, cls.psf_size),
+        )
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        del cls.multiple_cell_coadd
+        del cls.exposures
+        super().tearDownClass()
+
+
+class ExplodedCoaddTestCase(MultipleCellCoaddTestCase):
+    @classmethod
+    def setUpClass(cls) -> None:
+        super().setUpClass()
+        cls.exploded_coadd = cls.multiple_cell_coadd.explode()
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        del cls.exploded_coadd
+        super().tearDownClass()
+
+    def test_exploded_psf_image(self):
+        """Show that psf_image sizes are absurd."""
```

Comment:
I included this test temporarily to show that `pad_psfs_with` is not doing anything sensible. I am not sure what the desired behavior is. It should either a boolean parameter, that makes the PSF size same as the inner bbox size, or pad the PSF image with an integer value (not `float`). Or is there a different use case that I am missing?

Reply:
I was pretty confused, too, at first, until I read pretty deep into the code - it looks like it's the value to set the pixels to when padding, so it should really always be zero or `None`.

Since we've discovered that that's super confusing, let's change the name to `zero_pad` and make it a boolean.  When that's true I think we'll actually need to pad it to the outer cell size, not the inner cell size, since the purpose of that is to make the PSF image grid line up with the exploded image grid.

That can all go on another ticket, of course - and I'm not actually certain we even care about maintaining `ExplodedCoadd` in the long term; it was intended to support serialization in that form, but if we're going to do serialization via tables (probably FITS binary tables, but maybe Parquet), we might not need it.

Reply:
I think `ExplodedCoadd` might be helpful for visualization/eye-balling purposes even if not to support FITS persistence. So I'm inclined to keep it, but without many added methods. I'll make the suggested changes in another ticket. I'll leave this test as it is now, and modify the test in that other ticket.